### PR TITLE
eng, prepare release 0.21.1

### PR DESCRIPTION
**Pull request by weidongxu-microsoft**
include sync of https://github.com/microsoft/typespec/pull/4535
it may require bump of customization lib and release of it.

### CODE REVIEW

**Comment by weidongxu-microsoft** [comment]
spotless timeout https://github.com/Azure/autorest.java/pull/2957/checks?check_run_id=30705958175

File: protocol-sdk-integration-tests/.vscode/eclipse-format-azure-sdk-for-java.xml
Comment (weidongxu-microsoft) [inline protocol-sdk-integration-tests/.vscode/eclipse-format-azure-sdk-for-java.xml:1]: changes in protocol-sdk-integration-tests is for syncing parent pom, formatting config, code-quality config from SDK repo.

File: protocol-sdk-integration-tests/eng/code-quality-reports/src/main/java/com/azure/tools/checkstyle/checks/ImmutableClassCheck.java
Comment (XiaofeiCao) [inline protocol-sdk-integration-tests/eng/code-quality-reports/src/main/java/com/azure/tools/checkstyle/checks/ImmutableClassCheck.java:1]: Are these checkstyle files synced from sdk repo?

**Review by XiaofeiCao — APPROVED** [review]

File: protocol-sdk-integration-tests/eng/code-quality-reports/src/main/java/com/azure/tools/checkstyle/checks/ImmutableClassCheck.java
Comment (weidongxu-microsoft) [reply protocol-sdk-integration-tests/eng/code-quality-reports/src/main/java/com/azure/tools/checkstyle/checks/ImmutableClassCheck.java:1]: Yes. I just copied the folder.

**Pull request merged**